### DISCUSSION TITLE
Pico with NRF24L01

### DISCUSSION THREAD

**Discussion by Spetru-bit** [micropython/micropython#11909] [RP2040 / Pico] [answered]
Hi ! I want to make a contoller with 2 joysticks, a NRF24L01 module and a Rasberry Pi Pico. I bought new modules tu be sure that they work and 2 adaptor boards. When they arrived I tested them with 2 Arduino UNOs and they worked. I switched to the Pico boards and it didn't work and I don't know why. Thanks !

**Comment by peterhinch** [comment]
Start out by running the official demo which has been tested with the Pico. Then adapt to meet your requirements.

**Reply by Spetru-bit** [reply]
If you refer to this https://github.com/micropython/micropython/blob/v1.19/drivers/nrf24l01/nrf24l01test.py, it doesn't work

**Reply by Spetru-bit** [reply]
it says hardware not responding but it doesn't make any sense because the modules are good

**Reply by peterhinch** [reply, marked as answer]
You are looking at outdated code. The current version is [here](https://github.com/micropython/micropython-lib/blob/master/micropython/drivers/radio/nrf24l01/nrf24l01test.py).

**Reply by Spetru-bit** [reply]
Thanks ! Now it worked but I get mixed results and I don't know why, depending if the modules are pointed at each other or not. But when I tested them with the UNOs it didn't matter the orientation

**Reply by peterhinch** [reply]
Some Chinese NRF24l01 modules are useless. I strongly recommend Sparkfun ones which actually work.

**Reply by Spetru-bit** [reply]
With arduino the range was better + sparkfun ones are only on their site and the shipping is more expensive than the modules

**Reply by Spetru-bit** [reply]
And I looked now and I can't find them in stock

**Reply by peterhinch** [reply]
Alas it seems the Sparkfun ones are a "retired" product. Perhaps someone else will know a source for decent ones. I've had Chinese ones that couldn't communicate further than a couple of metres.
